features/timeseries/src/main/java/org/opennms/netmgt/timeseries/integration/TimeseriesWriter.java

```diff
@@ -0,0 +1,298 @@
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.joda.time.Duration;
+import org.opennms.core.logging.Logging;
+import org.opennms.integration.api.v1.timeseries.StorageException;
+import org.opennms.integration.api.v1.timeseries.Tag;
+import org.opennms.integration.api.v1.timeseries.immutables.ImmutableMetric;
+import org.opennms.integration.api.v1.timeseries.immutables.ImmutableSample;
+import org.opennms.integration.api.v1.timeseries.immutables.ImmutableTag;
+import org.opennms.netmgt.timeseries.impl.TimeseriesStorageManager;
+import org.opennms.netmgt.timeseries.meta.MetaData;
+import org.opennms.netmgt.timeseries.meta.TimeSeriesMetaDataDao;
+import org.opennms.newts.api.MetricType;
+import org.opennms.newts.api.Sample;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Preconditions;
+import com.google.common.math.DoubleMath;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.lmax.disruptor.EventTranslatorOneArg;
+import com.lmax.disruptor.FatalExceptionHandler;
+import com.lmax.disruptor.RingBuffer;
+import com.lmax.disruptor.WorkHandler;
+import com.lmax.disruptor.WorkerPool;
+import com.swrve.ratelimitedlogger.RateLimitedLog;
+
+/**
+ * Used to write samples to the {@link org.opennms.newts.api.SampleRepository}.
+ *
+ * Calls to  publish the samples to a ring buffer so
+ * that they don't block while the data is being persisted.
+ *
+ * @author jwhite
+ */
+public class TimeseriesWriter implements WorkHandler<SampleBatchEvent>, DisposableBean {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimeseriesWriter.class);
+
+    private static final RateLimitedLog RATE_LIMITED_LOGGER = RateLimitedLog
+            .withRateLimit(LOG)
+            .maxRate(5).every(Duration.standardSeconds(30))
+            .build();
+
+    private WorkerPool<SampleBatchEvent> workerPool;
+
+    private RingBuffer<SampleBatchEvent> ringBuffer;
+
+
+    private final int ringBufferSize;
+
+    private final int numWriterThreads;
+
+    private final Meter droppedSamples;
+
+    @Autowired
+    private TimeseriesStorageManager storage;
+
+    @Autowired
+    private TimeSeriesMetaDataDao timeSeriesMetaDataDao;
+
+    /**
+     * The {@link RingBuffer} doesn't appear to expose any methods that indicate the number
+     * of elements that are currently "queued", so we keep track of them with this atomic counter.
+     */
+    private final AtomicLong numEntriesOnRingBuffer = new AtomicLong();
+
+    @Inject
+    public TimeseriesWriter(@Named("timeseries.max_batch_size") Integer maxBatchSize, @Named("timeseries.ring_buffer_size") Integer ringBufferSize,
```

Comment:
Given that a lot of these tunables are the same as Newts, I'm wondering if we're better off just using the existing Newts system property names rather than defining a separate set.